src/object.cpp

```diff
@@ -30,3 +30,19 @@ void StageNode::Object::init()
   model->Subscribe();
   initialized_ = true;
 }
+
+void StageNode::Object::set_pose_rel(const std::shared_ptr<const Vehicle>& vehicle, const Stg::Pose rel_pose)
+{
+  double glob_x = vehicle->positionmodel->GetGlobalPose().x + std::cos(vehicle->positionmodel->GetGlobalPose().a) * rel_pose.x - std::sin(vehicle->positionmodel->GetGlobalPose().a) * rel_pose.y;
+  double glob_y = vehicle->positionmodel->GetGlobalPose().y + std::sin(vehicle->positionmodel->GetGlobalPose().a) * rel_pose.x + std::cos(vehicle->positionmodel->GetGlobalPose().a) * rel_pose.y;
+  double glob_yaw = vehicle->positionmodel->GetGlobalPose().a + rel_pose.a;
+  this->model->SetPose(Stg::Pose(glob_x, glob_y, rel_pose.z, glob_yaw));
+}
+
+void StageNode::Object::update_pose()
```

Comment:
I would put this in the stage_node.cpp instead. Reason is that it depends on the ROS node (although we save it in this object - not sure why). Since this class is basically an addition to stage wrapper I'd try decouple this thing a bit from the ROS stuff.